Packages/com.chisel.editor/Chisel/Editor/Components/Base/Editor/ChiselNodeDetails.cs

```diff
@@ -0,0 +1,45 @@
+﻿using UnityEngine;
+using UnityEditor;
+using UnityEditor.SceneManagement;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Chisel;
+using System.Reflection;
+using Chisel.Core;
+using Chisel.Components;
```

Comment:
ChiselGeneratorDetails is basically the meta data for one of our components. At the moment it only allows you to specific the HierarchyIcon, but we can extend it by adding Tool Icons and other stuff